### ISSUE TITLE
Search api, Display field copy, Form Mode Manager and Workflow patches

### ISSUE THREAD

**Issue by giancarlobi** [esmero/archipelago-deployment#99]
@DiegoPino I just open this to take trace of this question.
After installed Drupal 9.1 I see that :
- search api has  8.x-1.19 Stable release covered by the Drupal Security Team released 1 February 2021 that probably includes patch patches/1.18-3178941-metrocustom-referenced_entities_reindexing_fatal_error-D8.patch
- display field copy has  2.0.0 Stable release covered by the Drupal Security Team released 11 October 2020  that report Requires Drupal: ^8 || ^9 so probably the patch "D9 readyness": "https://www.drupal.org/files/issues/2020-10-01/drupal9-readiness_3135922-7.patch" is included while I don't know if also the second patch "Fix missing schema 1.x": "patches/1.x-display_field_copy-missing-schema-3107615-2-D8.patch",
All this derived from composer.json here https://github.com/esmero/archipelago-deployment/blob/1.0.0-RC2D9/composer.json

**Comment by giancarlobi** [comment]
@DiegoPino A first update. I installed  those modules on my fresh Drupal 9.1.5 without error. I don't have Archipelago modules yet, so check is partially. Anyway I can configure Solr connector and indexing. Take care friend

**Comment by giancarlobi** [comment]
@DiegoPino regarding form_mode_manager I can install it following partially this post: https://www.computerminds.co.uk/articles/apply-drupal-9-compatibility-patches-composer, that is:
- add to composer.json source package from git 
``` json
    },
    "repositories": [
    {
      "type": "package",
      "package": {
        "name": "drupal/form_mode_manager",
        "type": "drupal-module",
        "version": "dev-8.x-2.x",
        "source": {
          "type": "git",
          "url": "https://git.drupalcode.org/project/form_mode_manager.git",
          "reference": "bb634122"
        }
      }
    },
```
- add to composer.json entry for patches
``` json
        "enable-patching": true,
        "patches": {
            "drupal/form_mode_manager": {
                "Fix Form Mode title": "https://www.drupal.org/files/issues/2019-12-10/form_mode_manager-bundle-label-in-title-3100112-2-D8.patch",
                "D9 readyness": "https://www.drupal.org/files/issues/2020-10-30/form_mode_manager-drupal-9-compatibility-3164478-6.patch"
            },
```
- finally composer allows to install and apply patches
``` bash
$ composer -v require drupal/form_mode_manager
...
Installs: drupal/form_mode_manager:dev-8.x-2.x bb634122
...
  - Applying patches for drupal/form_mode_manager
...
$ vendor/bin/drush en form_mode_manager
```

**Comment by DiegoPino** [comment]
yes! That is similar to how i got it working (well similar, I had the bb634122 in the composer.lock for 9.1.x)
@pcambra we should ask the maintainer of that Module to allow us to co-maintain? Pretty desolated and untouched.

**Comment by giancarlobi** [comment]
@DiegoPino (no rush) do we need workflow module, right? I tried install 'drupal/workflow:^1.4' but when I enable it I receive this error: The service "workflow.route_subscriber" has a dependency on a non-existent service "entity.manager".
I searched and found this https://www.drupal.org/project/workflow/issues/3157199#comment-13980544 not so clear to me if patch solve or not. 
Currently I disabled and removed package.
Have you any more notes about this?

**Comment by DiegoPino** [comment]
@giancarlobi https://www.drupal.org/files/issues/2020-07-07/deprecated_entity_manager_workflowstorage_error-3157199-7.patch works fine. I "pinged" the maintainer.

**Comment by giancarlobi** [comment]
Thanks @DiegoPino . I added this to composer.json:
``` json
        "patches": {
            "drupal/workflow": {
                "D9 readyness": "https://www.drupal.org/files/issues/2020-07-07/deprecated_entity_manager_workflowstorage_error-3157199-7.patch"
            },
```
Install and enable all works well without errors.
